osDiscovery/osDiscovery.go

```diff
@@ -102,27 +103,43 @@ func GetDistributionRelease() (string, string, error) {
 // GetArchitecture returns a machine hardware name.
 func GetArchitecture() (string, error) {
 	if out, err := stubUtils.ExecCommand("uname", "-m"); err == nil {
-		return string(out), nil
+		out := string(out)
+		if strings.HasSuffix(out, "\n") {
+			out = out[:len(out)-1]
+		}
+		return out, nil
 	}
 	return "", ErrUnknownArchitecture
 }
 
 // GetKernel returns a kernel release.
 func GetKernel() (string, error) {
 	if out, err := stubUtils.ExecCommand("uname", "-r"); err == nil {
-		return string(out), nil
+		out := string(out)
+		if strings.HasSuffix(out, "\n") {
+			out = out[:len(out)-1]
+		}
+		return out, nil
```

Comment:
You can shorten all this by using the builtin `TrimSuffix` function. That function contains the exact same code but is easier to the eye.